### PR TITLE
Scrollable within a div

### PR DESCRIPTION
**Pull request by olegshilov**
#17

implement scrolling in custom container using element id.

### CODE REVIEW

**Comment by fisshy** [comment]
Thanks for the PR. I will look into it and test it. But why would you remove ReactDOM? I think it's better to follow the updates and if you have to use the older version you just need to downgrade.

**Comment by olegshilov** [comment]
@fisshy sorry, maybe removing ReactDOM is unnecessary. I can't make it work in my project with react 0.13.3. Something was wrong with peerDependency.

**Comment by ntdb** [comment]
+1 for this feature (scrolling inside a container) and +1 for the problems with ReactDOM/0.13.3. I'm using react-scroll 0.22.0 right now with 0.13.3 due to the peerDependency issue.

**Comment by fisshy** [comment]
Will probably merge this before the end of this week - haven't had the time to check it out yet.

**Comment by mbgjohansson** [comment]
@fisshy, do you think you will find time to look into this solution of scrolling within a div anytime soon?

**Comment by fisshy** [comment]
I will most likely look into this PR and see if I can merge it. Or I will accept any other PR with tests written.

**Comment by nathanielkornet** [comment]
would love to be able to use this!

File: build/npm/lib/mixins/Helpers.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ var Helpers = {
 
     propTypes: {
       to: React.PropTypes.string.isRequired,
+      containerId: React.PropTypes.string,
```
Comment (dapi) [inline build/npm/lib/mixins/Helpers.js]: Why it is not required?

File: build/npm/lib/mixins/Helpers.js [hunk 2]
```diff
@@ -67,23 +69,25 @@ var Helpers = {
 
           if(!element) {
               element = scroller.get(to);
-
               var cords = element.getBoundingClientRect();
-              elemTopBound = (cords.top + y);
+              var containeRect = container.getBoundingClientRect();
```
Comment (dapi) [inline build/npm/lib/mixins/Helpers.js]: what if container is not found? It is common situation

**Comment by dapi** [comment]
Why don't use anchor in href link? It is standard.

May be better to rename the property from `containerId` to `anchor`. And looks not only for any elements with `id` but for `a[name=anchor]` elements also.

**Comment by daiheitan** [comment]
It's 7 months and I am still waiting for this PR to be merged...

**Comment by fisshy** [comment]
Agree, I've started to merge most of the PR's, this one will take a bit more time since the codebase has changed some and I'll have to write some tests for it and not break current code.

**Comment by dapi** [comment]
Does this PR resolves the goal? https://github.com/fisshy/react-scroll/pull/69/files

**Comment by daiheitan** [comment]
I cannot test it as I'm using `react@15.0.1`. Maybe the dependencies need an update too. @fisshy

**Comment by daiheitan** [comment]
I checked the PR, don't think it solves the problem. This lib hardcoded `window.scrollTop`, which is the reason why srollable inside a `div` isn't working. Your code didn't modify that. @dapi

**Comment by fisshy** [comment]
Yes should upgrade React version. And yes, you might be correct, it's the most likely reason it will not work if we don't somehow create it as a module. @daiheitan

**Comment by daiheitan** [comment]
Alright, I will try that, thanks

**Comment by daiheitan** [comment]
Btw, simply move `react` & `react-dom` to `peerDependencies` will do the trick and allow people using `npm@3` to use `react@15.0`, if you need a quick fix...

**Comment by fisshy** [comment]
Thanks @daiheitan. Will do that on my lunch break :P.  Or you can make a PR and I will publish ;)

**Comment by daiheitan** [comment]
XD Too lazy to PR, I will wait...

**Comment by daiheitan** [comment]
@fisshy I read the code. If I get your thought correctly, I need to provide a custom `scroller` and custom `Link` component to get scrolls inside a div. Also, if I need animated scroll, I need to implement my own `animate-scroll.js` functionality. That's almost half of the lib...

**Comment by fisshy** [comment]
@devfishy You could extend the animate-scroll if you need features in it. And maybe some parts could be optimized for sure and extracted out to other modules.

File: build/npm/lib/mixins/Helpers.js (hunk 1 above)
Comment (poacher2k) [inline build/npm/lib/mixins/Helpers.js]: I'm guessing because it should default to `window` if not specified.

**Comment by ztratar** [comment]
This PR would make react-scroll *much* more valuable. 10 months since the initial PR here... bump?

**Comment by poacher2k** [comment]
@ztratar: I recently implemented this for a client project. Not tested extensively, but used in production with good result. You can see an example [here](http://oiw.no/program) with a device that has > 1000px width. The dates on the left side uses "spy" and scrolls the container that the event listings is in.

If you want, I could publish this here on GitHub - newer versions of NPM makes it easy to reference forks in package.json, so you can include it without any issues.

**Comment by fisshy** [comment]
It's actually my fault. If anyone could fix the PR to match current code base I'm more then happy to merge it. It's to far ahead now.

**Comment by poacher2k** [comment]
@fisshy: I added the functionality to [this fork](https://github.com/poacher2k/react-scroll). If it looks good, I can re-edit the README and make a PR.

**Comment by poacher2k** [comment]
I'll do a another fork and do things properly, without adding my own beautification and indentations.

**Comment by poacher2k** [comment]
Added PR #96

**Comment by cersfeld** [comment]
hi @poacher2k, I think I might have found a bug in your PR (in animate-scroll.js): in line 174, the function `scroleMore` calls the function `currenPositionY`. But at this point the variable `__containerElement` (line 63) is always undefined and the y-position it not calculated correctly.

**Comment by poacher2k** [comment]
Good find, @cersfeld, I'll look at this directly after some breakfast.

Den tor. 7. jul. 2016, 08:08 skrev cersfeld <notifications@github.com>:

**Comment by juannorris** [comment]
For anyone coming to this, the feature is working in the current version (1.5.2), just make sure the container div is positioned (e.g., `position: relative`).

Maybe the docs can be updated to explain that (I apologize if it's already there an I missed it).

@fisshy thanks for the library, it saved me a lot of time.

**Comment by olegshilov** [comment]
Ok. I'm close this because i think this PR is not actual anymore.

**Pull request closed without merging**